test/integration/error_output.cc

```diff
@@ -219,3 +222,103 @@ TEST(ErrorOutput, ElementErrorOutput)
 }
+
+////////////////////////////////////////
+// Test Model class for sdf::Errors outputs
+TEST(ErrorOutput, ModelErrorOutput)
+{
+  std::stringstream buffer;
+  sdf::testing::RedirectConsoleStream redir(
+    sdf::Console::Instance()->GetMsgStream(), &buffer);
+
+  sdf::Errors errors;
+
+  std::function findFileCb = [](const std::string &_uri)
+  {
+    return sdf::testing::TestFile("integration", "model", _uri);
+  };
+
+  std::ostringstream stream;
+  stream << "<?xml version=\"1.0\"?>"
+         << "<sdf version='1.8'>"
+         << "  <model name='test_model'>"
+         << "    <include>"
+         << "      <uri>test_model</uri>"
+         << "      <name>common_name</name>"
+         << "    </include>"
+         << "    <link name='common_name'/>"
+         << "    <link name='common_name'/>"
+         << "    <joint name='common_name' type='fixed'>"
+         << "       <parent>world</parent>"
+         << "       <child>child</child>"
+         << "    </joint>"
+         << "    <joint name='common_name' type='fixed'>"
+         << "       <parent>world</parent>"
+         << "       <child>child</child>"
+         << "    </joint>"
+         << "    <frame name='common_name'/>"
+         << "    <frame name='common_name'/>"
+         << "  </model>"
+         << "</sdf>";
+
+  sdf::SDFPtr sdfParsed(new sdf::SDF());
+  sdf::init(sdfParsed);
+
+  sdf::ParserConfig parserConfig;
+  parserConfig.SetWarningsPolicy(sdf::EnforcementPolicy::ERR);
+  parserConfig.SetFindCallback(findFileCb);
+  sdf::readString(stream.str(), parserConfig, sdfParsed, errors);
+  EXPECT_TRUE(errors.empty());
+
+  sdf::Model model;
+  errors = model.Load(sdfParsed->Root()->GetElement("model"), parserConfig);
+  ASSERT_EQ(errors.size(), 7u);
+  EXPECT_NE(std::string::npos, errors[0].Message().find(
+    "Non-unique name[common_name] detected 7 times in XML children of model"
+    " with name[test_model]."));
+  EXPECT_NE(std::string::npos, errors[1].Message().find(
+    "link with name[common_name] already exists."));
+  EXPECT_NE(std::string::npos, errors[2].Message().find(
+    "Link with name [common_name] in model with name [test_model] has a name"
+    " collision. Please rename this link."));
+  EXPECT_NE(std::string::npos, errors[3].Message().find(
+    "joint with name[common_name] already exists."));
+  EXPECT_NE(std::string::npos, errors[4].Message().find(
+    "Joint with name [common_name] in model with name [test_model] has a name"
+    " collision. Please rename this joint."));
+  EXPECT_NE(std::string::npos, errors[5].Message().find(
+    "frame with name[common_name] already exists."));
+  EXPECT_NE(std::string::npos, errors[6].Message().find(
+    "Frame with name [common_name] in model with name [test_model] has a name"
+    " collision. Please rename this frame."));
+  errors.clear();
+
+  //Set SDF version to someting lower than 1.7 for extra errors
+  sdfParsed->SetOriginalVersion("1.6.0");
+  sdf::Model second_model;
```

Comment:
nit: camelCase: `second_model` -> `secondModel`

Reply:
Fixed, now reusing the same model variable, just clearing links, joints and frames between loads: https://github.com/gazebosim/sdformat/pull/1122/commits/cb8099b458d95fa9cc05fd6a843fa064fe97d459.

Reply:
I would advice against doing that because if we add anything down the road in `sdf::Model` that needs to be cleared, we'll have to remember to come here and update this. Instead I recommend putting `sdf::Model model` in a new scopes, so that it gets destructed every time, i.e.:
```
{
  sdf::Model model;
  errors = model.Load(...);
  // test 1
  ...
}
{
  sdf::Model model;
  errors = model.Load(...);
  // test 2
  ...
}
```